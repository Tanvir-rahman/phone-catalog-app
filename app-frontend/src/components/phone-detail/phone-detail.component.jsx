import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { phoneDetails } from '@store/phones/phone.selectors';

import './phone-detail.styles.scss';

export const PhoneDetailComponent = ({ details, history }) => {
  return (
    details && (<div className='phone-details'>
      <div className='title'>Details</div>
      <div className='details-info'>
        <div className='details-info-img'>
          <img
            src={require('../../assets/images/' + details.imageurl)}
            alt={details.title}
          />
        </div>
        <div className='details-info-description'>
          <p>
            <span>Title: </span>
            {details.title}
          </p>
          <p>
            <span>Screen: </span>
            {details.screen}
          </p>
          <p>
            <span>Capacity: </span>
            {details.capacity}
          </p>
          <p>
            <span>Ram: </span>
            {details.ram}
          </p>
          <p>
            <span>Color: </span>
            {details.color}
          </p>
          <p>
            <span>Price: </span>
            ${details.price}
          </p>
          <p>{details.description}</p>
        </div>
      </div>
    </div>)
  );
};

const mapStateToProps = createStructuredSelector({
  details: phoneDetails,
});

export default connect(mapStateToProps)(PhoneDetailComponent);
