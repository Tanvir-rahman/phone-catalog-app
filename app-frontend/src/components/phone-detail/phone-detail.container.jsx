import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { isPhoneDetailsFetching } from '@store/phones/phone.selectors';
import WithSpinner from '@components/with-spinner/with-spinner.component';
import PhoneDetailComponent from './phone-detail.component';

const mapStateToProps = createStructuredSelector({
  isLoading: isPhoneDetailsFetching
});

const PhoneDetailContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(PhoneDetailComponent);

export default PhoneDetailContainer;
