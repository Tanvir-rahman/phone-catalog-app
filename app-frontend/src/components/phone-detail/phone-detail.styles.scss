.phone-details {
  .details-info {
    display: flex;
    .details-info-img {
      width: 30%;
      img {
        width: 100%;
      } 
    }
    span {
      font-weight: bold;
    }
  }
}

@media (max-width: 600px) {
  .phone-details {
    .details-info {
      flex-direction: column;
      .details-info-img {
        width: 100%;
      }
    }
  }
}