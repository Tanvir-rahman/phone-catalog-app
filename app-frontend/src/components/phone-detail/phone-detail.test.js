import React from 'react';
import { shallow } from 'enzyme';

import { PhoneDetailComponent } from './phone-detail.component';

describe('PhoneDetailComponent component', () => {
  let wrapper;
  const mockId = 'iphone-11'
  const mockTitle = 'Iphone 11';
  const mockScreen = '6.1 inches';
  const mockCapacity = '64 GB';
  const mockRam = '4GB';
  const mockColor = 'Grey';
  const mockDescription = 'Apple iPhone 11 Â· Released 2019, September 20';
  const mockPrice = '100';
  const mockImageUrl = 'iphone-11.png';

  beforeEach(() => {
    const mockProps = {
      id: mockId,
      title: mockTitle,
      screen: mockScreen,
      capacity: mockCapacity,
      ram: mockRam,
      color: mockColor,
      description: mockDescription,
      price: mockPrice,
      imageurl: mockImageUrl
    };

    wrapper = shallow(<PhoneDetailComponent details={mockProps} />);
  });

  it('should render PhoneDetailComponent component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title prop in Name Container', () => {
    expect(wrapper.find('.details-title').text()).toBe(`Title: ${mockTitle}`);
  });

  it('should render screen prop in Name Container', () => {
    expect(wrapper.find('.details-screen').text()).toBe(`Screen: ${mockScreen}`);
  });

  it('should render capacity prop in Name Container', () => {
    expect(wrapper.find('.details-capacity').text()).toBe(`Capacity: ${mockCapacity}`);
  });

  it('should render ram prop in Name Container', () => {
    expect(wrapper.find('.details-ram').text()).toBe(`Ram: ${mockRam}`);
  });

  it('should render color prop in Name Container', () => {
    expect(wrapper.find('.details-color').text()).toBe(`Color: ${mockColor}`);
  });

  it('should render price prop in Name Container', () => {
    expect(wrapper.find('.details-price').text()).toBe(`Price: $${mockPrice}`);
  });
});
