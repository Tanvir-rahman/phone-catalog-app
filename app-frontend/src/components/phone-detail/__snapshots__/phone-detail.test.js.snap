// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDetailComponent component should render PhoneDetailComponent component 1`] = `
<div
  className="phone-details"
>
  <div
    className="title"
  >
    Details
  </div>
  <div
    className="details-info"
  >
    <div
      className="details-info-img"
    >
      <img
        alt="Iphone 11"
        src="iphone-11.png"
      />
    </div>
    <div
      className="details-info-description"
    >
      <p
        className="details-title"
      >
        <span>
          Title: 
        </span>
        Iphone 11
      </p>
      <p
        className="details-screen"
      >
        <span>
          Screen: 
        </span>
        6.1 inches
      </p>
      <p
        className="details-capacity"
      >
        <span>
          Capacity: 
        </span>
        64 GB
      </p>
      <p
        className="details-ram"
      >
        <span>
          Ram: 
        </span>
        4GB
      </p>
      <p
        className="details-color"
      >
        <span>
          Color: 
        </span>
        Grey
      </p>
      <p
        className="details-price"
      >
        <span>
          Price: 
        </span>
        $
        100
      </p>
      <p>
        Apple iPhone 11 Â· Released 2019, September 20
      </p>
    </div>
  </div>
</div>
`;
